### PR TITLE
parser: parse Unicode identifiers

### PR DESCRIPTION
**Pull request by belochub**
Refs: https://github.com/metarhia/jstp/issues/152

### CODE REVIEW

**Comment by belochub** [comment]
@aqrln, may you help me with creation of pre-build configuration script to make it possible to choose which of the two implementations to use (check out https://github.com/metarhia/jstp/commit/89427481f29d1c50b2cd6ec042493a251cef26cf#diff-9689e209b75f216133f2b377028d89f6R9)?

File: tools/make-unicode-tables.js [hunk 1]
```diff
@@ -0,0 +1,186 @@
...
+const UCD_LINK = 'http://www.unicode.org/Public/' + UNICODE_VERSION +
+  '/ucd/DerivedCoreProperties.txt';
+const fullTablesFilename = 'unicode_tables.h';
+const rangeTablesFilename = 'unicode_range_tables.h';
+const getHeaderGuard = filename =>
+  `SRC_${filename.replace(/\W/g, '_').toUpperCase()}_`;
+const getOutputPath = filename => path.join(__dirname, '../src', filename);
+
+const getFileHeader = filename =>
+`// Copyright (c) 2017 JSTP project authors. Use of this source code is
```
Comment (lundibundi) [inline tools/make-unicode-tables.js]: Maybe it'll be better to put license in a separate file?

Comment (belochub) [inline tools/make-unicode-tables.js]: In my opinion it is good to have it here, because it will probably help to avoid any legal problems, given the fact that this file is also a piece of software which works with Unicode data files. Consider this as a precaution, I don't know for sure but I also don't want to find out about it in a bad way.

It helps to understand the resulting format of the output files as well, so that you don't need to look anywhere else or open this auto-generated files (which are pretty big and can slow down some of the editors) to understand their contents.

Comment (lundibundi) [inline tools/make-unicode-tables.js]: Fine with me =), reasonable enough.

**Comment by aqrln** [comment]
@belochub can you please rebase to resolve conflicts?

File: tools/make-unicode-tables.js [hunk 2]
```diff
@@ -0,0 +1,186 @@
...
+const idContinueRanges = [];
+
+let idStartTotalCount = 0;
+let idContinueTotalCount = 0;
+
+const lineRegex = /^([0-9A-F]{4,6})(?:\.\.([0-9A-F]{4,6}))? *; (\w+) #.*$/;
+
+http.get(UCD_LINK, (res) => {
+  const linereader = byline.createStream(res);
+  linereader.on('data', (line) => {
```
Comment (aqrln) [inline tools/make-unicode-tables.js]: Well, wouldn't the core `readline` module be sufficient to accomplish this?

Comment (belochub) [inline tools/make-unicode-tables.js]: Yeah, it probably is sufficient, I missed that.

File: package-lock.json [hunk 3]
```diff
@@ -4637,8 +4637,7 @@
             "wordwrap": {
               "version": "0.0.3",
               "resolved": "https://registry.npmjs.org/wordwrap/-/wordwrap-0.0.3.tgz",
-              "integrity": "sha1-o9XabNXAvAAI03I0u68b7WMFkQc=",
-              "dev": true
```
Comment (aqrln) [inline package-lock.json]: Huh? Looks like npm correctly handles the case when a package becomes used in `dependencies` and not only in `devDependencies`, but does not handle the case when the package is not required for `dependencies` anymore. Can you please revert this change manually?

Comment (aqrln) [inline package-lock.json]: Oh, wait, I think you only modified `devDependencies`. Now I really wonder what happened here.

**Comment by belochub** [comment]
@aqrln, we can't land it yet, you still haven't addressed https://github.com/metarhia/jstp/pull/218#issuecomment-307757110.

**Comment by belochub** [comment]
@aqrln, ping.

**Comment by aqrln** [comment]
@belochub I've pushed a commit to your branch.
@nechaido @lundibundi PTAL

**Comment by aqrln** [comment]
Landed in 4e942ae06d0a291bea37a6c6ffb42642b9842d4f and 4b67cbdb2a50266fa5f6ab6f1760811fc29465e0.
